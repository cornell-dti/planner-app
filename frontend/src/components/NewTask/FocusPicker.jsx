// @flow strict

import React from 'react';
import { Icon } from 'semantic-ui-react';
import styles from './Picker.css';

type Props = {|
  +onPinChange: (inFocus: boolean) => void;
|};
type State = {| +pinned: boolean; |}

export default class FocusPicker extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { pinned: false };
  }

  handleOpenClose = (e: SyntheticEvent<HTMLElement>) => {
    e.stopPropagation();
    const { onPinChange } = this.props;
    const { pinned } = this.state;
    this.setState({ pinned: !pinned });
    onPinChange(!pinned);
  };

  resetState = () => this.setState({ pinned: false });

  render() {
    const { pinned } = this.state;
    return (
      <div className={styles.Main}>
        <span
          role="button"
          tabIndex={-1}
          onClick={this.handleOpenClose}
          onKeyDown={() => {}}
          style={{ background: 'none' }}
          className={styles.LabelHack}
        >
          <Icon
            name="pin"
            className={styles.CenterIcon}
            style={{ color: pinned ? 'black' : 'gray' }}
          />
        </span>
      </div>
    );
  }
}
