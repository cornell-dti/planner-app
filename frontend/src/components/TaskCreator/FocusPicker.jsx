// @flow strict

import React from 'react';
import Pin from '../../assets/svgs/pin-2-dark-outline.svg';
import styles from './Picker.css';

type Props = {|
  +pinned: boolean;
  +onPinChange: (inFocus: boolean) => void;
|};

export default function FocusPicker({ pinned, onPinChange }: Props) {
  const clickPicker = (e: SyntheticEvent<HTMLElement>) => {
    e.stopPropagation();
    onPinChange(!pinned);
  };
  const iconStyle = { /* color: pinned ? 'black' : 'gray' */
    backgroundImage: `url(${pinned ? Pin : Pin})`,
  };
  return (
    <div className={styles.Main}>
      <span role="presentation" onClick={clickPicker} className={styles.Label}>
        <Pin className={styles.CenterIcon} />
      </span>
    </div>
  );
}
