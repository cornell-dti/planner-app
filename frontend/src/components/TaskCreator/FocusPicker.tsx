import React, { ReactElement, SyntheticEvent, KeyboardEvent } from 'react';
import clsx from 'clsx';
import styles from './Picker.module.scss';
import SamwiseIcon from '../UI/SamwiseIcon';

type Props = {
  readonly pinned: boolean;
  readonly onPinChange: (inFocus: boolean) => void;
};

export default function FocusPicker({ pinned, onPinChange }: Props): ReactElement {
  const clickPicker = (e: SyntheticEvent<HTMLElement>): void => {
    e.stopPropagation();
    onPinChange(!pinned);
  };
  const pressedPicker = (e: KeyboardEvent): void => {
    if (e.key === 'Enter' || e.key === ' ') {
      e.stopPropagation();
      onPinChange(!pinned);
    }
  };
  const pinIconName = pinned ? 'pin-dark-filled' : 'pin-dark-outline';
  return (
    <div className={styles.Main}>
      <button
        type="button"
        onClick={clickPicker}
        onKeyPress={pressedPicker}
        className={clsx(styles.TestBorder, styles.Label)}
      >
        <SamwiseIcon iconName={pinIconName} className={clsx(styles.CenterIcon, styles.ImageSize)} />
      </button>
    </div>
  );
}
