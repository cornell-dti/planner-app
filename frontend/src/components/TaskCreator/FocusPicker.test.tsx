import React from 'react';
import renderer from 'react-test-renderer';
import FocusPicker from './FocusPicker';

it('Pinned FocusPicker matches snapshot', () => {
  const tree = renderer.create(
    <FocusPicker pinned onPinChange={() => { }} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Unpinned FocusPicker matches snapshot', () => {
  const tree = renderer.create(
    <FocusPicker pinned={false} onPinChange={() => { }} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
