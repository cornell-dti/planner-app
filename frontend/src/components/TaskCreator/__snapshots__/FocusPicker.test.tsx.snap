// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pinned FocusPicker matches snapshot 1`] = `
<div
  className="Main"
>
  <span
    className="Label"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <span
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
      title="Unpin from Focus"
    >
      <img
        alt="Unpin from Focus"
        className="CenterIcon"
        src="/svgs/pin-2-dark-filled.svg"
        style={
          Object {
            "height": "1em",
            "width": "1em",
          }
        }
      />
    </span>
  </span>
</div>
`;

exports[`Unpinned FocusPicker matches snapshot 1`] = `
<div
  className="Main"
>
  <span
    className="Label"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <span
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
      title="Pin to Focus"
    >
      <img
        alt="Pin to Focus"
        className="CenterIcon"
        src="/svgs/pin-2-dark-outline.svg"
        style={
          Object {
            "height": "1em",
            "width": "1em",
          }
        }
      />
    </span>
  </span>
</div>
`;
